### ISSUE TITLE
Google Cloud Monitoring plugin didn't load QueryType

### ISSUE THREAD

**Issue by kylewin** [grafana/grafana#91532]
### What happened?

Dashboard Settings > Variables > Make a new Variable with Datasource of Google Cloud Monitoring
It shows "Loading"
<img width="528" alt="image" src="https://github.com/user-attachments/assets/906a0692-88c3-4586-80a5-2fd5d41c102e">

### What did you expect to happen?

The plugin should show QueryType values
<img width="574" alt="image" src="https://github.com/user-attachments/assets/1951086c-25b0-43ca-96b7-54bde2e8c678">

### Did this work before?

Yes it did

### How do we reproduce it?

1. Dashboard settings Cog icon
2. Variables tab
3. Create a new Variable with Google Cloud Monitoring datasource
4. Click on Query Type

### Environment (with versions)?

Grafana: 11.1.3
OS: Sonoma 14.5
Browser: Chrome Version 127.0.6533.74 (Official Build) (arm64)

### Grafana platform?

Kubernetes

### Datasource(s)?

Google Cloud Monitoring

**Comment by mazemax** [comment]
For me these options are working for the previously created dashboard, but not the new one.
It looks like a small effort UI issue.
CC: @aangelisc @asimpson @andresmgot

**Comment by aangelisc** [comment]
Hi @kylewin, @mazemax,

I've tested this with our 11.1.3 docker image and the latest and I am unable to replicate it with both new and existing dashboards.

Do you have any network logs showing failed requests that might indicate why this is happening?

**Comment by kylewin** [comment]
I have this. But this is too weird

![image](https://github.com/user-attachments/assets/067ba4d0-f13e-455c-8257-bf44f9e39e9a)

This error even shows up on my working Dashboard (using the same Datasource). If I import the dashboards from this Datasource, everything work fine. It's still fine in Grafana Explore. Only if I make a new Variable, the Plugin cannot load values

**Comment by kylewin** [comment]
> If you run the health check on your data source does it pass fine with no errors? Are you also able to query fine?
Yes, everything is fine with the datasource. As I mentioned above, when I import the Dashboard from Datasource screen, everything work fine. Only when I add a new variable, the plugin will not be able to show QueryType

<img width="796" alt="image" src="https://github.com/user-attachments/assets/5484ce8d-b172-4192-bc46-e7db1da61c0a">

These dashboards work fine:
<img width="511" alt="image" src="https://github.com/user-attachments/assets/b03ebe94-b7e8-4078-8914-15e2d6e172fb">

**Comment by aangelisc** [comment]
Hi @kylewin, how is your data source authenticated and do you see an error in the HTTP response that may give more information as to why the request is forbidden?

**Comment by aangelisc** [comment]
Hi @kylewin, closing this for now but if the issue reoccurs please re-open and let us know 😊

**Comment by lboutin-mwm** [comment]
This is still an issue in Grafana 12.3.0

The Data Source has sufficient permissions for fetching data within panels and is valid when tested directly through the data source panel but is unable to do anything but load within Settings -> Variables

<img width="887" height="266" alt="Image" src="https://github.com/user-attachments/assets/74278169-7dd8-4aff-b738-b7d0f0fb4a69" />

<img width="805" height="115" alt="Image" src="https://github.com/user-attachments/assets/835e2807-be31-4f6d-8f00-b9cffbd27fed" />

<img width="526" height="700" alt="Image" src="https://github.com/user-attachments/assets/3a53969a-c4a8-473c-9b07-275ab7a17445" />

This specific token is scoped to a singular project within GCP, so it might be an issue related to looking for other projects and not having enough rights to do so ?
